### ISSUE TITLE
GetLineNumberFromPosition exception

### ISSUE THREAD

**Issue by EmilAlipiev** [mrlacey/Rapid-XAML-Toolkit#396]
**Describe the bug**
I am getting below exception every time. Any idea?

```
Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Error	RXT999	Unexpected error occurred while parsing XAML.
Views\StartPage.xaml	1		Please log an issue to https://github.com/mrlacey/Rapid-XAML-Toolkit/issues Reason: System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
Parameter name: position
   at Microsoft.VisualStudio.Text.Implementation.BinaryStringRebuilder.GetLineNumberFromPosition(Int32 position)
   at Microsoft.VisualStudio.Text.Implementation.CachingTextImage.GetLineNumberFromPosition(Int32 position)
   at Microsoft.VisualStudio.Text.Implementation.BaseSnapshot.GetLineFromPosition(Int32 position)
   at RapidXamlToolkit.XamlAnalysis.Tags.RapidXamlDisplayedTag..ctor(TagDependencies deps, String errorCode, TagErrorType defaultErrorType)
   at RapidXamlToolkit.XamlAnalysis.Tags.CustomAnalysisTag..ctor(CustomAnalysisTagDependencies deps)
   at RapidXamlToolkit.XamlAnalysis.CustomProcessorWrapper.Process(String fileName, Int32 offset, String xamlElement, String linePadding, ITextSnapshot snapshot, TagList tags, List`1 suppressions)
   at RapidXamlToolkit.XamlAnalysis.XamlElementExtractor.Parse(ProjectType projectType, String fileName, ITextSnapshot snapshot, String xaml, List`1 processors, TagList tags, IVisualStudioAbstraction vsAbstraction, List`1 suppressions, String projectFilePath)
   at RapidXamlToolkit.XamlAnalysis.RapidXamlDocument.Create(ITextSnapshot snapshot, String fileName, IVisualStudioAbstraction vsa, String projectFile)
```

**Comment by mrlacey** [comment]
Hi @EmilAlipiev 
what are you doing when you get this?
Which version of VisualStudio are you using?
Can you share the contents of `StartPage.xaml`?

**Comment by mrlacey** [comment]
Version 0.11 has just been submitted to the marketplace and it includes code to avoid this problem throwing unhandled errors and to capture more information about the problem if/when it happens again.

Please can you try updating the extension to the new version and then reporting what happens.

**Comment by mrlacey** [comment]
Ok. I think I know why this is happening.
working on it

**Comment by mrlacey** [comment]
Version 0.11.3 is now in the marketplace.
It includes some code to try and address this problem. 
However, as I haven't been able to fully recreate the issue, it also includes additional information in error reports to try and identify the cause if the changes don't fix it.

I'll leave this open until the fix is verified or more information is available to identify the cause.

**Comment by mrlacey** [comment]
It's taken ages to track down (sorry) by I now know the cause of this--and how to fix it.

**Comment by mrlacey** [comment]
This functionality has been released as part of [v 0.12.0 and is available now](https://marketplace.visualstudio.com/items?itemName=MattLaceyLtd.RapidXamlToolkit).
